server/pypi/make.sh

```diff
@@ -1,57 +1,285 @@
+              --year year"
+      exit 0
+      ;;
+    --package)
+      PACKAGES=${2}
+      shift
+      shift
+      ;;
+    --skip-deps)
+      BUILD_DEPS="0"
+      shift
+      ;;
+    --skip-envs)
+      BUILD_ENVS="0"
+      shift
+      ;;
+    --year)
+      YEAR=${2}
+      shift
+      shift
+      ;;
+    *)
+      POSITIONAL+=("$1") # save it in an array for later
+      shift
+    ;;
+  esac
+done
+
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+# setup clean conda environments and toolchains for each Python versions
+
+eval "$(command conda 'shell.bash' 'hook')"
+conda activate base
+
+if [ "${BUILD_ENVS}" == "1" ]; then
+  rm -rf "${LOGS}" "${TOOLCHAINS}"
+  curl --silent --location "https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-macos-universal.tar.gz" --output cmake.tar.gz
+  mkdir -p "${TOOLCHAINS}"
+  tar -xzf cmake.tar.gz --directory "${TOOLCHAINS}"
+  mv "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"*/CMake.app "${TOOLCHAINS}"
+  rm -rf "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"* cmake.tar.gz
+
+  for PYTHON_VERSION in ${PYTHON_VERSIONS}; do
+    rm -rf "$(conda info | grep 'envs directories' | awk -F ':' '{ print $2 }' | sed -e 's/^[[:space:]]*//')/${CONDA_ENV:?}"
+    conda create -y --name "${CONDA_ENV}-${PYTHON_VERSION}" "python==${PYTHON_EXACT_VERSIONS[${PYTHON_VERSION}]}"
+    conda activate "${CONDA_ENV}-${PYTHON_VERSION}"
+    pip3 install -r requirements.txt
+    curl --silent --location "$(curl --silent -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" --location https://api.github.com/repos/beeware/Python-Apple-support/releases | grep browser_download_url | grep iOS | grep "${PYTHON_VERSION}" | head -1 | awk '{ print $2 }' | tr -d '"')" --output "python-apple-support-${PYTHON_VERSION}.tar.gz"
+    mkdir -p "${TOOLCHAINS}/${PYTHON_VERSION}" "${LOGS}/${PYTHON_VERSION}"
+    tar -xzf "python-apple-support-${PYTHON_VERSION}.tar.gz" --directory "${TOOLCHAINS}/${PYTHON_VERSION}"
+    rm -rf "python-apple-support-${PYTHON_VERSION}.tar.gz" "${LOGS:?}/${PYTHON_VERSION:?}"/*
+  done
+fi
+
+# build dependencies
+
+if [ "${BUILD_DEPS}" == "1" ]; then
+  mkdir -p "${LOGS}/deps"
+  pushd "${TOOLCHAINS}"
+
+  if ! [ -d "${PYTHON_APPLE_SUPPORT}" ]; then
+    git clone https://github.com/beeware/Python-Apple-support.git
+  fi
+
+  conda activate "${CONDA_ENV}-3.10"
+  pushd "${PYTHON_APPLE_SUPPORT}"
+  git checkout 3.10
+  make libFFI-wheels
+  # TODO: make OpenSSL-wheels
+  make BZip2-wheels
+  make XZ-wheels
+  popd
+  popd
+
+  mkdir -p "${DIST_DIR}"
+  rm -rf "${DIST_DIR}/bzip2" "${DIST_DIR}/libffi" "${DIST_DIR}/xz"
+  mv -f "${TOOLCHAINS}/${PYTHON_APPLE_SUPPORT}/wheels/dist"/* "${DIST_DIR}"
+
+  for DEPENDENCY in ${DEPENDENCIES}; do
+    printf "\n\n*** Building dependency %s ***\n\n" "${DEPENDENCY}"
+    python build-wheel.py --toolchain "${TOOLCHAINS}" --python 3.10 iOS "${DEPENDENCY}" 2>&1 | tee "${LOGS}/deps/${DEPENDENCY}.log"
```

Comment:
It took me a moment to work out why the hard-code version of Python; once I worked out that it was for dependencies only, it made more sense. However, the 3.10 version will need to be updated at some point in the future, and will need to be updated in multiple places because of being hard coded. 

Also - following an earlier comment - the hard-coding becomes unnecessary if the script is made "single Python version", as the dependencies pass can be done using whatever Python is in the build environment. In terms of parallelisation, that means we would only need to do a dependencies pass on *one* Python version, then do all the other Python versions (or do "just dependencies" on on Python version, and then "packages" on all versions)